app/assets/javascripts/infinity_scroll.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+function init_infinity_scroll(initial_page_number, request_url, load_first_page) {
+    var page_number;
+    if (initial_page_number === undefined) {
+        page_number = 1;
+    }
+    else {
+        page_number = initial_page_number
+    }
+    if (load_first_page === true) load_data();
+    start_infinity_scroll();
+
+    function start_infinity_scroll() {
+        $(window).on('scroll', function () {
+            if (end_of_page()) {
+                load_data();
+                $(window).off('scroll');
+            }
+        });
+    }
+
+    function end_of_page() {
+        return $(window).scrollTop() + $(window).height() > $(document).height() - 100
+    }
+
+    function load_data() {
```

Comment:
This script was written for comments, and refactored now. I've seen this gems, but they didn`t suite me, because they are related to pagination block. I had a lot of troubles with this, because comments are polymorphic and they doesn't have itself page.
I've decided to write my own script.
About gems, I think gem's are good, because they increase speed of development. But for learning programming better to write native code, am I right?